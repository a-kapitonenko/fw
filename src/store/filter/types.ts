import { Frame } from '../frames/types';

export type Field = {
  name: string;
  value: string;
  checked?: boolean;
};

export type Groups = {
  color: Field[];
  width: Field[];
  noseBridge: Field[];
  shape: Field[];
  material: Field[];
};

export const enum FilterActionTypes {
  FETCH_REQUEST = '@@filter/FETCH_REQUEST',
  FETCH_SUCCESS = '@@filter/FETCH_SUCCESS',
  FETCH_ERROR = '@@filter/FETCH_ERROR',
  CHANGE_CHECKED = '@@filter/CHANGE_CHECKED',
};

export interface IFilterState {
  readonly fetching: boolean;
  readonly groups: Groups;
  readonly errors: string;
  readonly query: Object;
  readonly frames: Frame[];
};
