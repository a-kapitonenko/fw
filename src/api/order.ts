import request from './request';
// import { ApplicationState } from '../store';
import { Boss, Prescription,  } from '../store/order/types';
import { saveOrderInformation, saveOrder, savePrescription, saveFittingHeight, loadOrderValues } from '../test/order';

export const submitRequest = (data: Boss) => request(saveOrderInformation, data);
export const saveOrderRequest = (data: any) => request(saveOrder, data);
export const fetchOrderValuesRequest = (data: string) => request(loadOrderValues, data);
export const savePrescriptionRequest = (data: Prescription) => request(savePrescription, data)
export const saveFittingHeightRequest = (data: { boss: Boss, height: number }) => request(saveFittingHeight, data);
