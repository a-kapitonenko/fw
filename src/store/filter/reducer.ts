import { Reducer } from 'redux';
import { IFilterState, FilterActionTypes, Groups, Field } from './types';

const initialGroupsState = {
  color: [],
  width: [],
  noseBridge: [],
  shape: [],
  material: []
}

const initialState: IFilterState = {
  isFetching: false,
  errors: '',
  query: initialGroupsState,
  frames: [],
  groups: {
    isFetching: false,
    errors: '',
    data: <Groups>initialGroupsState,
  }
};

const reducer: Reducer<IFilterState> = (state = initialState, action) => {
  switch (action.type) {
    case FilterActionTypes.FILTERING_START: {
      return { ...state, isFetching: true, errors: '' };
    }
    case FilterActionTypes.FILTERING_SUCCESS: {
      return { ...state, isFetching: false, frames: action.payload };
    }
    case FilterActionTypes.FILTERING_FAILED: {
      return { ...state, isFetching: false, errors: action.payload };
    }
    case FilterActionTypes.FETCH_GROUPS_START: {
      return { ...state, groups: { ...state.groups, isFetching: true, errors: '' } };
    }
    case FilterActionTypes.FETCH_GROUPS_SUCCESS: {
      return { ...state, groups: { ...state.groups, isFetching: false, data: action.payload } };
    }
    case FilterActionTypes.FETCH_GROUPS_FAILED: {
      return { ...state, groups: { ...state.groups, isFetching: true, errors: action.paylaod } };
    }
    case FilterActionTypes.CHANGE_CHECKED: {
      const item = state.groups.data[action.payload.type].map((field: Field) => {
        if (field.name === action.payload.name) {
          return { ...field, checked: action.payload.value };
        }

        return { ...field };
      });

      return {
        ...state,
        groups: {
          ...state.groups,
          data: {
            ...state.groups.data,
            [action.payload.type]: item
          }
        }
      };
    }
    case FilterActionTypes.CLEAR_CHECKED: {
      const groups = <Groups>{};

      for (const group in state.groups.data) {
        groups[group] = state.groups.data[group].map((field: Field) => ({ ...field, checked: false }));
      }

      return { ...state, groups: { ...state.groups, data: groups } };
    }
    case FilterActionTypes.ADD_QUERY: {
      return {
        ...state,
        query: {
          ...state.query,
          [action.payload.type]: [
            ...state.query[action.payload.type],
            action.payload.value
          ]
        }
      };
    }
    case FilterActionTypes.DELETE_QUERY: {
      const index = state.query[action.payload.type].indexOf(action.payload.value);

      return {
        ...state,
        query: {
          ...state.query,
          [action.payload.type]: [
            ...state.query[action.payload.type].slice(0, index),
            ...state.query[action.payload.type].slice(index + 1)
          ]
        }
      };
    }
    case FilterActionTypes.CLEAR_QUERY: {
      return { ...state, query: initialGroupsState };
    }
    default: {
      return state;
    }
  }
};

export { reducer as FilterReducer };
