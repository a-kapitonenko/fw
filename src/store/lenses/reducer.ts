import { Reducer } from 'redux';
import { LensesState, LensesActionTypes } from './types';

const initialState: LensesState = {
  fetching: false,
  list: [],
  errors: '',
}

const reducer: Reducer<LensesState> = (state = initialState, action) => {
  switch (action.type) {
    case LensesActionTypes.FETCH_REQUEST: {
      return { ...state, fetching: true };
    }
    case LensesActionTypes.FETCH_SUCCESS: {
      return { 
        ...state, 
        fetching: false, 
        list: action.payload 
      };
    }
    case LensesActionTypes.FETCH_ERROR: {
      return { 
        ...state, 
        fetching: false, 
        errors: action.payload 
      };
    }
    default: {
      return state
    }
  }
};

export { reducer as LensesReducer }
