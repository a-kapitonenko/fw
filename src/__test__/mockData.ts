import { ApplicationState } from '../store';
import { OculusInfo, Prescription, Boss, Blueprint, Barcode, Errors } from '../store/order/types';
import { Frame, FramesState } from '../store/frames/types';
import { Groups, IFilterState } from '../store/filter/types';
import { Lens } from '../store/lenses/types';
import { initialValue } from '../test/database';
import { RouterState } from 'react-router-redux';
import { ISearchState } from '../store/search/types';

export const errorMessage: string = 'Error message';
export const recommendation: string = 'Based on the prescription, the lens selection will use an 8 base lenses that is 70 MM in diameter. We recommend you look for frames with a strong front curve and a small Frame ED (<60 ED)';
export const message = 'The selected parameters for frame and lens are acceptable for cotout.';
export const checkLensErrorMessage: string = "Selected lens doesn't compatibility with patient prescription. Please, select from the following lenses";
export const checkFrameErrorMessage: string = 'This lens does not accecptable with selected frame.';
export const valuesErrorMessage: string = 'Failed to load order values';

const oculusData: OculusInfo = {
  sphere: '1',
  cyclinder: '2',
  axis: '1',
  addition: '2',
  PDDistance: '1',
  PDNear: '1',
  prism: '1',
};

export const prescription: Prescription = {
  OD: oculusData,
  OS: oculusData,
};

export const lens: Lens = {
  name: 'Nikon Eyes* Advanced + Polycarbonate- Polarized Gray/Brown',
  value: 1,
};

export const lenses: Lens[] = [
  {
    name: 'Nikon Eyes* Advanced + Polycarbonate- Polarized Gray/Brown',
    value: 1
  },
  {
    name: 'Nikon Eyes* Advanced + Polycarbonate- Transitions* Signature VII',
    value: 2
  },
  {
    name: 'Nikon Eyes* Advanced + Polycarbonate- XTRActive* Gray',
    value: 3
  },
];

export const frame: Frame = {
  upc: 82850553738,
  label: 'FM17116',
  value: '82850553738',
  img: '1.png',
  compatibility: true,
};

export const frames: Frame[] = [
  {
    upc: 82850553738,
    label: 'FM17116',
    value: '82850553738',
    img: '1.png',
    compatibility: true,
  },
  {
    upc: 82850523999,
    label: 'SA17346',
    value: '82850523999',
    img: '2.png',
    compatibility: false,
  },
];

export const groups: Groups = {
  color: [
    { name: 'Black', value: 'black' },
  ],
  width: [
    { name: 'Narrow', value: 'narrow' },
  ],
  noseBridge: [
    { name: 'Standard', value: 'standard' },
  ],
  shape: [
    { name: 'Square', value: 'square' },
  ],
  material: [
    { name: 'Acetate', value: 'acetate' },
  ],
};

export const query: object = {
  color: ['black', 'gold', 'brown'],
  width: ['narrow', 'medium'],
  noseBridge: ['standard'],
  shape: ['square'],
  material: ['acetate'],
};

export const fittingHeight = [
  {
    name: '1',
    value: 1,
  },
  {
    name: '2',
    value: 2,
  },
];

export const blueprint: Blueprint = {
  img: '1.png',
};

export const barcode: Barcode = {
  img: '1.png',
};

export const boss: Boss = {} as Boss;
export const state: ApplicationState = {} as ApplicationState;
export const id: string = '6E8C896C14153342DE34BBF26F69A8135BA548864A8C838F035CB35A593C2399';
export const values: any = initialValue;
export const height: number = 3;

export const filledState: ApplicationState = {
  routing: {} as RouterState,
  frames: {} as FramesState,
  lenses: {
    isFetching: false,
    errors: '',
    lenses: lenses,
  },
  order: {
    isFetching: false,
    errors: <Errors>{},
    redirect: false,
    fittingProperties: [],
    recommendation: '',
    message: '',
    blueprint: <Blueprint>{},
    boss: <Boss>{
      prescription: prescription,
      fittingHeight: 0,
      frame: frame,
      lens: lens,
      barcode: <Barcode>{},
    },
  },
  filter: {} as IFilterState,
  search: {} as ISearchState,
};
