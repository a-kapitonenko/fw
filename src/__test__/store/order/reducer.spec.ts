import { OrderActionTypes, Blueprint, BossTypes } from '../../../store/order/types';
import { OrderReducer } from '../../../store/order/reducer';
import * as mockData from '../../mockData';

describe('order reducer', () => {
  it('test handles SUBMIT_START', () => {
    const orderAction = {
      type: OrderActionTypes.SUBMIT_START,
      payload: {},
    };
    
    expect(OrderReducer(undefined, orderAction)).toMatchSnapshot();
  });

  it('test handles SUBMIT_SUCCESS', () => {
    const orderAction = {
      type: OrderActionTypes.SUBMIT_SUCCESS,
      payload: {},
    };
    
    expect(OrderReducer(undefined, orderAction)).toMatchSnapshot();
  });

  it('test handles SUBMIT_FAILED', () => {
    const orderPayload: string = mockData.errorMessage;

    const orderAction = {
      type: OrderActionTypes.SUBMIT_FAILED,
      payload: orderPayload,
    };
    
    expect(OrderReducer(undefined, orderAction)).toMatchSnapshot();
  });

  it('test handles SAVE_ORDER_START', () => {
    const orderAction = {
      type: OrderActionTypes.SAVE_ORDER_START,
      payload: {},
    };
    
    expect(OrderReducer(undefined, orderAction)).toMatchSnapshot();
  });

  it('test handles SAVE_ORDER_SUCCESS', () => {
    const orderAction = {
      type: OrderActionTypes.SAVE_ORDER_SUCCESS,
      payload: {},
    };
    
    expect(OrderReducer(undefined, orderAction)).toMatchSnapshot();
  });

  it('test handles SAVE_ORDER_FAILED', () => {
    const orderPayload: string = mockData.errorMessage;

    const orderAction = {
      type: OrderActionTypes.SAVE_ORDER_FAILED,
      payload: orderPayload,
    };
    
    expect(OrderReducer(undefined, orderAction)).toMatchSnapshot();
  });

  it('test handles FETCH_ORDER_VALUES_START', () => {
    const orderAction = {
      type: OrderActionTypes.FETCH_ORDER_VALUES_START,
      payload: {},
    };
    
    expect(OrderReducer(undefined, orderAction)).toMatchSnapshot();
  });

  it('test handles FETCH_ORDER_VALUES_SUCCESS', () => {
    const orderPayload: any = mockData.values;

    const orderAction = {
      type: OrderActionTypes.FETCH_ORDER_VALUES_SUCCESS,
      payload: orderPayload,
    };
    
    expect(OrderReducer(undefined, orderAction)).toMatchSnapshot();
  });

  it('test handles FETCH_ORDER_VALUES_FAILED', () => {
    const orderPayload: string = mockData.errorMessage;

    const orderAction = {
      type: OrderActionTypes.FETCH_ORDER_VALUES_FAILED,
      payload: orderPayload,
    };
    
    expect(OrderReducer(undefined, orderAction)).toMatchSnapshot();
  });

  it('test handles SAVE_PRESCRIPTION_START', () => {
    const orderAction = {
      type: OrderActionTypes.SAVE_PRESCRIPTION_START,
      payload: {},
    };
    
    expect(OrderReducer(undefined, orderAction)).toMatchSnapshot();
  });

  it('test handles SAVE_PRESCRIPTION_SUCCESS', () => {
    const orderAction = {
      type: OrderActionTypes.SAVE_PRESCRIPTION_SUCCESS,
      payload: {},
    };
    
    expect(OrderReducer(undefined, orderAction)).toMatchSnapshot();
  });

  it('test handles SAVE_PRESCRIPTION_FAILED', () => {
    const orderPayload: string = mockData.errorMessage;

    const orderAction = {
      type: OrderActionTypes.SAVE_PRESCRIPTION_FAILED,
      payload: orderPayload,
    };
    
    expect(OrderReducer(undefined, orderAction)).toMatchSnapshot();
  });

  it('test handles SAVE_FITTING_HEIGHT_START', () => {
    const orderAction = {
      type: OrderActionTypes.SAVE_FITTING_HEIGHT_START,
      payload: {},
    };
    
    expect(OrderReducer(undefined, orderAction)).toMatchSnapshot();
  });

  it('test handles SAVE_FITTING_HEIGHT_SUCCESS', () => {
    const orderPayload: Blueprint = mockData.blueprint;

    const orderAction = {
      type: OrderActionTypes.SAVE_FITTING_HEIGHT_SUCCESS,
      payload: orderPayload,
    };
    
    expect(OrderReducer(undefined, orderAction)).toMatchSnapshot();
  });

  it('test handles SAVE_FITTING_HEIGHT_FAILED', () => {
    const orderPayload: string = mockData.errorMessage;

    const orderAction = {
      type: OrderActionTypes.SAVE_FITTING_HEIGHT_FAILED,
      payload: orderPayload,
    };
    
    expect(OrderReducer(undefined, orderAction)).toMatchSnapshot();
  });

  it('test handles SET_RX_INFORMATION', () => {
    const orderPayload: { type: string, field: string, value: string } = {
      type: 'OD',
      field: 'sphere',
      value: '-7.5',
    };

    const orderAction = {
      type: OrderActionTypes.SET_RX_INFORMATION,
      payload: orderPayload,
    };
    
    expect(OrderReducer(undefined, orderAction)).toMatchSnapshot();
  });

  it('test handles SET_RECOMMENDATION', () => {
    const orderPayload: string = mockData.recommendation;

    const orderAction = {
      type: OrderActionTypes.SET_RECOMMENDATION,
      payload: orderPayload,
    };
    
    expect(OrderReducer(undefined, orderAction)).toMatchSnapshot();
  });

  it('test handles SET_MESSAGE', () => {
    const orderPayload: string = mockData.message;

    const orderAction = {
      type: OrderActionTypes.SET_MESSAGE,
      payload: orderPayload,
    };
    
    expect(OrderReducer(undefined, orderAction)).toMatchSnapshot();
  });

  it('test handles SET_FITTING_PROPERTIES', () => {
    const orderPayload: any = mockData.fittingHeight;

    const orderAction = {
      type: OrderActionTypes.SET_FITTING_PROPERTIES,
      payload: orderPayload,
    };
    
    expect(OrderReducer(undefined, orderAction)).toMatchSnapshot();
  });

  it('test handles SET_BOSS', () => {
    const orderPayload: { type: BossTypes, value: any } = {
      type: BossTypes.LENS,
      value: mockData.lens,
    };
    
    const orderAction = {
      type: OrderActionTypes.SET_BOSS,
      payload: orderPayload,
    };
    
    expect(OrderReducer(undefined, orderAction)).toMatchSnapshot();
  });

  it('test DISABLE_REDIRECT SET_BOSS', () => {
    const orderAction = {
      type: OrderActionTypes.DISABLE_REDIRECT,
      payload: {},
    };
    
    expect(OrderReducer(undefined, orderAction)).toMatchSnapshot();
  });
});
