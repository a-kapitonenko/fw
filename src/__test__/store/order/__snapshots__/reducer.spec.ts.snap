// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`order reducer test DISABLE_REDIRECT SET_BOSS 1`] = `
Object {
  "blueprint": Object {},
  "boss": Object {
    "barcode": Object {},
    "fittingHeight": 0,
    "frame": Object {},
    "lens": Object {},
    "prescription": Object {
      "OD": Object {
        "PDDistance": "",
        "PDNear": "",
        "addition": "",
        "axis": "",
        "cyclinder": "",
        "prism": "",
        "sphere": "",
      },
      "OS": Object {
        "PDDistance": "",
        "PDNear": "",
        "addition": "",
        "axis": "",
        "cyclinder": "",
        "prism": "",
        "sphere": "",
      },
    },
  },
  "errors": Object {},
  "fittingProperties": Array [],
  "isFetching": false,
  "message": "",
  "recommendation": "",
  "redirect": false,
}
`;

exports[`order reducer test handles FETCH_ORDER_VALUES_FAILED 1`] = `
Object {
  "blueprint": Object {},
  "boss": Object {
    "barcode": Object {},
    "fittingHeight": 0,
    "frame": Object {},
    "lens": Object {},
    "prescription": Object {
      "OD": Object {
        "PDDistance": "",
        "PDNear": "",
        "addition": "",
        "axis": "",
        "cyclinder": "",
        "prism": "",
        "sphere": "",
      },
      "OS": Object {
        "PDDistance": "",
        "PDNear": "",
        "addition": "",
        "axis": "",
        "cyclinder": "",
        "prism": "",
        "sphere": "",
      },
    },
  },
  "errors": Object {
    "submit": "Error message",
  },
  "fittingProperties": Array [],
  "isFetching": false,
  "message": "",
  "recommendation": "",
  "redirect": false,
}
`;

exports[`order reducer test handles FETCH_ORDER_VALUES_START 1`] = `
Object {
  "blueprint": Object {},
  "boss": Object {
    "barcode": Object {},
    "fittingHeight": 0,
    "frame": Object {},
    "lens": Object {},
    "prescription": Object {
      "OD": Object {
        "PDDistance": "",
        "PDNear": "",
        "addition": "",
        "axis": "",
        "cyclinder": "",
        "prism": "",
        "sphere": "",
      },
      "OS": Object {
        "PDDistance": "",
        "PDNear": "",
        "addition": "",
        "axis": "",
        "cyclinder": "",
        "prism": "",
        "sphere": "",
      },
    },
  },
  "errors": Object {
    "submit": "",
  },
  "fittingProperties": Array [],
  "isFetching": true,
  "message": "",
  "recommendation": "",
  "redirect": false,
}
`;

exports[`order reducer test handles FETCH_ORDER_VALUES_SUCCESS 1`] = `
Object {
  "blueprint": 0,
  "boss": Object {
    "barcode": Object {},
    "fittingHeight": 0,
    "frame": Object {
      "compatibility": true,
      "img": "4.png",
      "label": "FN17116",
      "upc": "52850523738",
      "value": "52850523738",
    },
    "lens": Object {
      "name": "Nikon Eyes* Affinity Trivex-Clear",
      "value": 4,
    },
    "prescription": Object {
      "OD": Object {
        "PDDistance": "1",
        "PDNear": "1",
        "addition": "11",
        "axis": "11",
        "cyclinder": "1",
        "prism": "11",
        "sphere": "1",
      },
      "OS": Object {
        "PDDistance": "1",
        "PDNear": "1",
        "addition": "11",
        "axis": "11",
        "cyclinder": "1",
        "prism": "11",
        "sphere": "1",
      },
    },
  },
  "errors": Object {},
  "fittingProperties": Array [
    Object {
      "name": "1",
      "value": 1,
    },
    Object {
      "name": "2",
      "value": 2,
    },
    Object {
      "name": "3",
      "value": 3,
    },
    Object {
      "name": "4",
      "value": 4,
    },
    Object {
      "name": "5",
      "value": 5,
    },
  ],
  "isFetching": false,
  "message": "The selected parameters for frame and lens are acceptable for cotout.",
  "recommendation": "Based on the prescription, the lens selection will use an 8 base lenses that is 70 MM in diameter. We recommend you look for frames with a strong front curve and a small Frame ED (<60 ED)",
  "redirect": false,
}
`;

exports[`order reducer test handles SAVE_FITTING_HEIGHT_FAILED 1`] = `
Object {
  "blueprint": Object {},
  "boss": Object {
    "barcode": Object {},
    "fittingHeight": 0,
    "frame": Object {},
    "lens": Object {},
    "prescription": Object {
      "OD": Object {
        "PDDistance": "",
        "PDNear": "",
        "addition": "",
        "axis": "",
        "cyclinder": "",
        "prism": "",
        "sphere": "",
      },
      "OS": Object {
        "PDDistance": "",
        "PDNear": "",
        "addition": "",
        "axis": "",
        "cyclinder": "",
        "prism": "",
        "sphere": "",
      },
    },
  },
  "errors": Object {
    "fittingHeight": "Error message",
  },
  "fittingProperties": Array [],
  "isFetching": false,
  "message": "",
  "recommendation": "",
  "redirect": false,
}
`;

exports[`order reducer test handles SAVE_FITTING_HEIGHT_START 1`] = `
Object {
  "blueprint": Object {},
  "boss": Object {
    "barcode": Object {},
    "fittingHeight": 0,
    "frame": Object {},
    "lens": Object {},
    "prescription": Object {
      "OD": Object {
        "PDDistance": "",
        "PDNear": "",
        "addition": "",
        "axis": "",
        "cyclinder": "",
        "prism": "",
        "sphere": "",
      },
      "OS": Object {
        "PDDistance": "",
        "PDNear": "",
        "addition": "",
        "axis": "",
        "cyclinder": "",
        "prism": "",
        "sphere": "",
      },
    },
  },
  "errors": Object {
    "fittingHeight": "",
  },
  "fittingProperties": Array [],
  "isFetching": true,
  "message": "",
  "recommendation": "",
  "redirect": false,
}
`;

exports[`order reducer test handles SAVE_FITTING_HEIGHT_SUCCESS 1`] = `
Object {
  "blueprint": Object {
    "img": "1.png",
  },
  "boss": Object {
    "barcode": Object {},
    "fittingHeight": 0,
    "frame": Object {},
    "lens": Object {},
    "prescription": Object {
      "OD": Object {
        "PDDistance": "",
        "PDNear": "",
        "addition": "",
        "axis": "",
        "cyclinder": "",
        "prism": "",
        "sphere": "",
      },
      "OS": Object {
        "PDDistance": "",
        "PDNear": "",
        "addition": "",
        "axis": "",
        "cyclinder": "",
        "prism": "",
        "sphere": "",
      },
    },
  },
  "errors": Object {},
  "fittingProperties": Array [],
  "isFetching": false,
  "message": "",
  "recommendation": "",
  "redirect": false,
}
`;

exports[`order reducer test handles SAVE_ORDER_FAILED 1`] = `
Object {
  "blueprint": Object {},
  "boss": Object {
    "barcode": Object {},
    "fittingHeight": 0,
    "frame": Object {},
    "lens": Object {},
    "prescription": Object {
      "OD": Object {
        "PDDistance": "",
        "PDNear": "",
        "addition": "",
        "axis": "",
        "cyclinder": "",
        "prism": "",
        "sphere": "",
      },
      "OS": Object {
        "PDDistance": "",
        "PDNear": "",
        "addition": "",
        "axis": "",
        "cyclinder": "",
        "prism": "",
        "sphere": "",
      },
    },
  },
  "errors": Object {
    "submit": "Error message",
  },
  "fittingProperties": Array [],
  "isFetching": false,
  "message": "",
  "recommendation": "",
  "redirect": false,
}
`;

exports[`order reducer test handles SAVE_ORDER_START 1`] = `
Object {
  "blueprint": Object {},
  "boss": Object {
    "barcode": Object {},
    "fittingHeight": 0,
    "frame": Object {},
    "lens": Object {},
    "prescription": Object {
      "OD": Object {
        "PDDistance": "",
        "PDNear": "",
        "addition": "",
        "axis": "",
        "cyclinder": "",
        "prism": "",
        "sphere": "",
      },
      "OS": Object {
        "PDDistance": "",
        "PDNear": "",
        "addition": "",
        "axis": "",
        "cyclinder": "",
        "prism": "",
        "sphere": "",
      },
    },
  },
  "errors": Object {
    "submit": "",
  },
  "fittingProperties": Array [],
  "isFetching": true,
  "message": "",
  "recommendation": "",
  "redirect": false,
}
`;

exports[`order reducer test handles SAVE_ORDER_SUCCESS 1`] = `
Object {
  "blueprint": Object {},
  "boss": Object {
    "barcode": Object {},
    "fittingHeight": 0,
    "frame": Object {},
    "lens": Object {},
    "prescription": Object {
      "OD": Object {
        "PDDistance": "",
        "PDNear": "",
        "addition": "",
        "axis": "",
        "cyclinder": "",
        "prism": "",
        "sphere": "",
      },
      "OS": Object {
        "PDDistance": "",
        "PDNear": "",
        "addition": "",
        "axis": "",
        "cyclinder": "",
        "prism": "",
        "sphere": "",
      },
    },
  },
  "errors": Object {},
  "fittingProperties": Array [],
  "isFetching": false,
  "message": "",
  "recommendation": "",
  "redirect": false,
}
`;

exports[`order reducer test handles SAVE_PRESCRIPTION_FAILED 1`] = `
Object {
  "blueprint": Object {},
  "boss": Object {
    "barcode": Object {},
    "fittingHeight": 0,
    "frame": Object {},
    "lens": Object {},
    "prescription": Object {
      "OD": Object {
        "PDDistance": "",
        "PDNear": "",
        "addition": "",
        "axis": "",
        "cyclinder": "",
        "prism": "",
        "sphere": "",
      },
      "OS": Object {
        "PDDistance": "",
        "PDNear": "",
        "addition": "",
        "axis": "",
        "cyclinder": "",
        "prism": "",
        "sphere": "",
      },
    },
  },
  "errors": Object {
    "prescription": "Error message",
  },
  "fittingProperties": Array [],
  "isFetching": false,
  "message": "",
  "recommendation": "",
  "redirect": false,
}
`;

exports[`order reducer test handles SAVE_PRESCRIPTION_START 1`] = `
Object {
  "blueprint": Object {},
  "boss": Object {
    "barcode": Object {},
    "fittingHeight": 0,
    "frame": Object {},
    "lens": Object {},
    "prescription": Object {
      "OD": Object {
        "PDDistance": "",
        "PDNear": "",
        "addition": "",
        "axis": "",
        "cyclinder": "",
        "prism": "",
        "sphere": "",
      },
      "OS": Object {
        "PDDistance": "",
        "PDNear": "",
        "addition": "",
        "axis": "",
        "cyclinder": "",
        "prism": "",
        "sphere": "",
      },
    },
  },
  "errors": Object {
    "prescription": "",
  },
  "fittingProperties": Array [],
  "isFetching": true,
  "message": "",
  "recommendation": "",
  "redirect": false,
}
`;

exports[`order reducer test handles SAVE_PRESCRIPTION_SUCCESS 1`] = `
Object {
  "blueprint": Object {},
  "boss": Object {
    "barcode": Object {},
    "fittingHeight": 0,
    "frame": Object {},
    "lens": Object {},
    "prescription": Object {
      "OD": Object {
        "PDDistance": "",
        "PDNear": "",
        "addition": "",
        "axis": "",
        "cyclinder": "",
        "prism": "",
        "sphere": "",
      },
      "OS": Object {
        "PDDistance": "",
        "PDNear": "",
        "addition": "",
        "axis": "",
        "cyclinder": "",
        "prism": "",
        "sphere": "",
      },
    },
  },
  "errors": Object {},
  "fittingProperties": Array [],
  "isFetching": false,
  "message": "",
  "recommendation": "",
  "redirect": false,
}
`;

exports[`order reducer test handles SET_BOSS 1`] = `
Object {
  "blueprint": Object {},
  "boss": Object {
    "barcode": Object {},
    "fittingHeight": 0,
    "frame": Object {},
    "lens": Object {
      "name": "Nikon Eyes* Advanced + Polycarbonate- Polarized Gray/Brown",
      "value": 1,
    },
    "prescription": Object {
      "OD": Object {
        "PDDistance": "",
        "PDNear": "",
        "addition": "",
        "axis": "",
        "cyclinder": "",
        "prism": "",
        "sphere": "",
      },
      "OS": Object {
        "PDDistance": "",
        "PDNear": "",
        "addition": "",
        "axis": "",
        "cyclinder": "",
        "prism": "",
        "sphere": "",
      },
    },
  },
  "errors": Object {},
  "fittingProperties": Array [],
  "isFetching": false,
  "message": "",
  "recommendation": "",
  "redirect": false,
}
`;

exports[`order reducer test handles SET_FITTING_PROPERTIES 1`] = `
Object {
  "blueprint": Object {},
  "boss": Object {
    "barcode": Object {},
    "fittingHeight": 0,
    "frame": Object {},
    "lens": Object {},
    "prescription": Object {
      "OD": Object {
        "PDDistance": "",
        "PDNear": "",
        "addition": "",
        "axis": "",
        "cyclinder": "",
        "prism": "",
        "sphere": "",
      },
      "OS": Object {
        "PDDistance": "",
        "PDNear": "",
        "addition": "",
        "axis": "",
        "cyclinder": "",
        "prism": "",
        "sphere": "",
      },
    },
  },
  "errors": Object {},
  "fittingProperties": Array [
    Object {
      "name": "1",
      "value": 1,
    },
    Object {
      "name": "2",
      "value": 2,
    },
  ],
  "isFetching": false,
  "message": "",
  "recommendation": "",
  "redirect": false,
}
`;

exports[`order reducer test handles SET_MESSAGE 1`] = `
Object {
  "blueprint": Object {},
  "boss": Object {
    "barcode": Object {},
    "fittingHeight": 0,
    "frame": Object {},
    "lens": Object {},
    "prescription": Object {
      "OD": Object {
        "PDDistance": "",
        "PDNear": "",
        "addition": "",
        "axis": "",
        "cyclinder": "",
        "prism": "",
        "sphere": "",
      },
      "OS": Object {
        "PDDistance": "",
        "PDNear": "",
        "addition": "",
        "axis": "",
        "cyclinder": "",
        "prism": "",
        "sphere": "",
      },
    },
  },
  "errors": Object {},
  "fittingProperties": Array [],
  "isFetching": false,
  "message": "The selected parameters for frame and lens are acceptable for cotout.",
  "recommendation": "",
  "redirect": false,
}
`;

exports[`order reducer test handles SET_RECOMMENDATION 1`] = `
Object {
  "blueprint": Object {},
  "boss": Object {
    "barcode": Object {},
    "fittingHeight": 0,
    "frame": Object {},
    "lens": Object {},
    "prescription": Object {
      "OD": Object {
        "PDDistance": "",
        "PDNear": "",
        "addition": "",
        "axis": "",
        "cyclinder": "",
        "prism": "",
        "sphere": "",
      },
      "OS": Object {
        "PDDistance": "",
        "PDNear": "",
        "addition": "",
        "axis": "",
        "cyclinder": "",
        "prism": "",
        "sphere": "",
      },
    },
  },
  "errors": Object {},
  "fittingProperties": Array [],
  "isFetching": false,
  "message": "",
  "recommendation": "Based on the prescription, the lens selection will use an 8 base lenses that is 70 MM in diameter. We recommend you look for frames with a strong front curve and a small Frame ED (<60 ED)",
  "redirect": false,
}
`;

exports[`order reducer test handles SET_RX_INFORMATION 1`] = `
Object {
  "blueprint": Object {},
  "boss": Object {
    "barcode": Object {},
    "fittingHeight": 0,
    "frame": Object {},
    "lens": Object {},
    "prescription": Object {
      "OD": Object {
        "PDDistance": "",
        "PDNear": "",
        "addition": "",
        "axis": "",
        "cyclinder": "",
        "prism": "",
        "sphere": "-7.5",
      },
      "OS": Object {
        "PDDistance": "",
        "PDNear": "",
        "addition": "",
        "axis": "",
        "cyclinder": "",
        "prism": "",
        "sphere": "",
      },
    },
  },
  "errors": Object {},
  "fittingProperties": Array [],
  "isFetching": false,
  "message": "",
  "recommendation": "",
  "redirect": false,
}
`;

exports[`order reducer test handles SUBMIT_FAILED 1`] = `
Object {
  "blueprint": Object {},
  "boss": Object {
    "barcode": Object {},
    "fittingHeight": 0,
    "frame": Object {},
    "lens": Object {},
    "prescription": Object {
      "OD": Object {
        "PDDistance": "",
        "PDNear": "",
        "addition": "",
        "axis": "",
        "cyclinder": "",
        "prism": "",
        "sphere": "",
      },
      "OS": Object {
        "PDDistance": "",
        "PDNear": "",
        "addition": "",
        "axis": "",
        "cyclinder": "",
        "prism": "",
        "sphere": "",
      },
    },
  },
  "errors": Object {
    "submit": "Error message",
  },
  "fittingProperties": Array [],
  "isFetching": false,
  "message": "",
  "recommendation": "",
  "redirect": false,
}
`;

exports[`order reducer test handles SUBMIT_START 1`] = `
Object {
  "blueprint": Object {},
  "boss": Object {
    "barcode": Object {},
    "fittingHeight": 0,
    "frame": Object {},
    "lens": Object {},
    "prescription": Object {
      "OD": Object {
        "PDDistance": "",
        "PDNear": "",
        "addition": "",
        "axis": "",
        "cyclinder": "",
        "prism": "",
        "sphere": "",
      },
      "OS": Object {
        "PDDistance": "",
        "PDNear": "",
        "addition": "",
        "axis": "",
        "cyclinder": "",
        "prism": "",
        "sphere": "",
      },
    },
  },
  "errors": Object {
    "submit": "",
  },
  "fittingProperties": Array [],
  "isFetching": true,
  "message": "",
  "recommendation": "",
  "redirect": false,
}
`;

exports[`order reducer test handles SUBMIT_SUCCESS 1`] = `
Object {
  "blueprint": Object {},
  "boss": Object {
    "barcode": Object {},
    "fittingHeight": 0,
    "frame": Object {},
    "lens": Object {},
    "prescription": Object {
      "OD": Object {
        "PDDistance": "",
        "PDNear": "",
        "addition": "",
        "axis": "",
        "cyclinder": "",
        "prism": "",
        "sphere": "",
      },
      "OS": Object {
        "PDDistance": "",
        "PDNear": "",
        "addition": "",
        "axis": "",
        "cyclinder": "",
        "prism": "",
        "sphere": "",
      },
    },
  },
  "errors": Object {},
  "fittingProperties": Array [],
  "isFetching": false,
  "message": "",
  "recommendation": "",
  "redirect": true,
}
`;
